mindmeld/components/domain_classifier.py

```diff
@@ -101,49 +105,50 @@ def inspect(self, query, domain=None, dynamic_resource=None):
         )
 
     def _get_query_tree(
-        self, queries=None, label_set=DEFAULT_TRAIN_SET_REGEX, raw=False
+        self, label_set=DEFAULT_TRAIN_SET_REGEX
     ):
         """Returns the set of queries to train on
 
         Args:
-            queries (list, optional): A list of ProcessedQuery objects, to
-                train. If not specified, a label set will be loaded.
             label_set (list, optional): A label set to load. If not specified,
                 the default training set will be loaded.
-            raw (bool, optional): When True, raw query strings will be returned
 
         Returns:
-            (list): list of queries
+            ProcessedQueryList: list of queries
         """
-        if queries:
-            return self._build_query_tree(queries, raw=raw)
-
-        return self._resource_loader.get_labeled_queries(label_set=label_set, raw=raw)
+        return self._resource_loader.get_labeled_queries(label_set=label_set)
 
-    def _get_queries_and_labels(self, queries=None, label_set=DEFAULT_TRAIN_SET_REGEX):
-        """Returns a set of queries and their labels based on the label set
+    def _get_queries_and_labels(self, label_set=DEFAULT_TRAIN_SET_REGEX):
+        """
+        Returns a set of queries and their labels based on the label set.
+        Because the query set is large for domain classification, it uses a
+        lazy loading iterator, the labels are relatively small and iterated
+        over multiple times so they are resolved to an in-memory list.
 
         Args:
-            queries (list, optional): A list of ProcessedQuery objects, to
-                train. If not specified, a label set will be loaded.
             label_set (list, optional): A label set to load. If not specified,
                 the default training set will be loaded.
+
+        Returns:
+            tuple(ProcessedQueryList.QueryIterator,list[str])
+
         """
-        query_tree = self._get_query_tree(queries, label_set=label_set)
+        query_tree = self._get_query_tree(label_set=label_set)
         queries = self._resource_loader.flatten_query_tree(query_tree)
 
         if not queries:
             return [None, None]
-        return list(zip(*[(q.query, q.domain) for q in queries]))
+        return (queries.queries(),
+                list(queries.domains()))
```

Comment:
Dont we want an iterator here to yield a tuple of (query, domain)?

Reply:
No, this is returning a tuple of iterators rather than an iterator of tuples.  This behaviour hasn't changed.

Reply:
For further clarity:
```
>>> list(zip(*[(i,i) for i in range(10)]))
[(0, 1, 2, 3, 4, 5, 6, 7, 8, 9), (0, 1, 2, 3, 4, 5, 6, 7, 8, 9)]
```